From c460b7bdf08fa3dc010b92e20c10d3926af94c53 Mon Sep 17 00:00:00 2001
From: Axel Gembe <derago@gmail.com>
Date: Thu, 27 Aug 2020 20:45:13 +0700
Subject: [PATCH] Windows: disable operator new in static build with jemalloc
 >= 5

Jemalloc 5 added its own operator new / delete which conflict with
the rocksdb ones when doing a static link.

See https://github.com/jemalloc/jemalloc/releases/tag/5.0.0

> Add C++ new/delete operator bindings. (@djwatson)

https://github.com/jemalloc/jemalloc/commit/2319152d9f5d9b33eebc36a50ccf4239f31c1ad9
---
 port/win/win_jemalloc.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/port/win/win_jemalloc.cc b/port/win/win_jemalloc.cc
index a29232f8a..9687da872 100644
--- a/port/win/win_jemalloc.cc
+++ b/port/win/win_jemalloc.cc
@@ -46,6 +46,8 @@ void jemalloc_aligned_free(void* p) ROCKSDB_NOEXCEPT { je_free(p); }
 }  // namespace port
 }  // namespace ROCKSDB_NAMESPACE
 
+#if defined(ROCKSDB_DLL) || (JEMALLOC_VERSION_MAJOR < 5)
+
 void* operator new(size_t size) {
   void* p = je_malloc(size);
   if (!p) {
@@ -73,3 +75,5 @@ void operator delete[](void* p) {
     je_free(p);
   }
 }
+
+#endif  // defined(ROCKSDB_DLL) || (JEMALLOC_VERSION_MAJOR < 5)
-- 
2.28.0.windows.1

